submissions/ORybka/frogger-game/app.js

```diff
@@ -0,0 +1,130 @@
+const param = {
+  fieldBorderX: [0, 400],
+  fieldBorderY: [0, 400],
+  enemyX: -50,
+  enemyY: [60, 145, 230],
+  minSpeed: 50,
+  maxSpeed: 200,
+  enemySize: 60,
+  initialX: 200,
+  initialY: 380,
+  playerWidth: 100,
+  playerHeight: 80,
+};
+
+const Enemy = function (x, y) {
+  this.sprite = 'images/enemy-bug.png';
+  this.x = x;
+  this.y = y;
+  this.speed = Math.floor(Math.random() * (param.maxSpeed - param.minSpeed) + param.minSpeed);
+  this.width = param.enemySize;
+  this.endField = param.fieldBorderX[1];
+  this.startX = param.enemyX;
+};
+
+const allEnemies = [];
+param.enemyY.forEach((el) => allEnemies.push(new Enemy(param.enemyX, el)));
+
+Enemy.prototype.update = function (dt) {
+  this.x += this.speed * dt;
+  if (this.x >= this.endField + this.width) {
+    this.x = this.startX;
+  }
+};
+
+Enemy.prototype.render = function () {
+  ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+const Player = function (x, y) {
+  this.sprite = 'images/char-cat-girl.png';
+  this.x = x;
+  this.y = y;
+  this.width = param.playerWidth;
+  this.height = param.playerHeight;
+  this.startFieldX = param.fieldBorderX[0];
+  this.endFieldX = param.fieldBorderX[1];
+  this.startFieldY = param.fieldBorderY[0];
+  this.endFieldY = param.fieldBorderY[1];
+  this.enemies = allEnemies;
```

Comment:
A class should be agnostic of the outer world directly. Whatever it needs to know about outer world should be explicitly given to it at construction time. Just like `x` and `y` for it to appear at.

`allEnemies` is somethig from the outer world. Fix this.

Although, it is OK if class refers to global constants (that is their purpose - to be global) and engine API (`Resources` in this case).